python/lsst/ts/m2com/mock/mock_command.py

```diff
@@ -1057,3 +1057,47 @@ async def set_configuration_file(
         await message_event.write_config()
 
         return model, CommandStatus.Success
+
+    async def set_hardpoint_list(
+        self, message: dict, model: MockModel, message_event: MockMessageEvent
+    ) -> tuple[MockModel, CommandStatus]:
+        """Set the hardpoint list.
+
+        Parameters
+        ----------
+        message : `dict`
+            Command message.
+        model : `MockModel`
+            Mock model to simulate the M2 hardware behavior.
+        message_event : `MockMessageEvent`
+            Instance of MockMessageEvent to write the event.
+
+        Returns
+        -------
+        model : `MockModel`
+            Mock model to simulate the M2 hardware behavior.
+        `CommandStatus`
+            Status of command execution.
+        """
+
+        # Set the hardpoints
+        control_open_loop = model.control_open_loop
+        lut_angle = (
+            model.inclinometer_angle_external
+            if model.control_parameters["use_external_elevation_angle"]
+            else control_open_loop.correct_inclinometer_angle(
+                control_open_loop.inclinometer_angle
+            )
+        )
+
+        model.control_closed_loop.update_hardpoints(message["actuators"], lut_angle)
+
+        # Send the event
+
+        # Note the index begins from 0 in Python
```

Comment:
In the LabVIEW, most of code is based on the 0-based actuator index. The Python mock code tries to mimic this behavior to avoid the confusion (and make my translation from LabVIEW to Python easier). The main reason here to change the index is for the following event in **ts_xml**:

```
  <SALEvent>
    <Subsystem>MTM2</Subsystem>
    <EFDB_Topic>MTM2_logevent_hardpointList</EFDB_Topic>
    <Description>Hardpoint list in the low-level controller.</Description>
    <item>
      <EFDB_Name>actuators</EFDB_Name>
      <Description>Actuator Ids to be the hardpoint.</Description>
      <IDL_Type>unsigned short</IDL_Type>
      <Units>unitless</Units>
      <Count>6</Count>
    </item>
  </SALEvent>
```

This is the 1-based actuator ID because Bo wanted to compare with the M2 configuration files in LabVIEW, which is 1-based unfortunately. For example, the following configuration file is 1-based actuator ID:

https://github.com/lsst-ts/ts_mtm2_cell/blob/develop/config/parameter_files/closed_loop_controller/CtrlParameterFiles_2018-07-19_104314_surg/Hd_ax_Matrix_Params.csv

Therefore, if we want to be consistent in the user's level, maybe I would prefer to modify the above event to be 0-based instead.

Thanks!

Reply:
It should be the rule to use human-rated arguments. Humans count from 1, computers from 0. M1M3 is using actuator IDs or hardpoint IDs, all are 1 based. I suggest to drop 0 based arguments.